zio-http/src/main/scala/zhttp/http/URL.scala

```diff
@@ -78,21 +78,23 @@ object URL {
     } yield url
   }
 
-  def asString(url: URL): String = {
-
-    def path = {
-      val encoder = new JQueryStringEncoder(url.path.asString)
-      url.queryParams.foreach { case (key, values) =>
-        values.foreach { value => encoder.addParam(key, value) }
-      }
-      encoder.toString
-    }
+  def asUriString(url: URL): String = path(url)
 
+  def asString(url: URL): String = {
+    val pathStr = path(url)
     url.kind match {
-      case Location.Relative                     => path
+      case Location.Relative                     => pathStr
       case Location.Absolute(scheme, host, port) =>
-        if (port == 80 || port == 443) s"${scheme.asString}://$host$path"
-        else s"${scheme.asString}://$host:$port$path"
+        if (port == 80 || port == 443) s"${scheme.asString}://$host$pathStr"
+        else s"${scheme.asString}://$host:$port$pathStr"
+    }
+  }
+
+  private def path(url: URL) = {
+    val encoder = new JQueryStringEncoder(url.path.asString)
+    url.queryParams.foreach { case (key, values) =>
+      values.foreach { value => encoder.addParam(key, value) }
     }
+    encoder.toString
   }
```

Comment:
Are these changes needed?

Reply:
That would depend on the comment above. If we need to use `uri` instead of `url` then I need a way of extracting a `uri` string. If we are okay to just use `url` then this wont be needed